src/tree/revisionManagement/RevisionDraftItem.ts

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { KnownActiveRevisionsMode, Revision } from "@azure/arm-appcontainers";
+import { TreeElementBase, nonNullProp } from "@microsoft/vscode-azext-utils";
+import type { AzureSubscription } from "@microsoft/vscode-azureresources-api";
+import { TreeItem, TreeItemCollapsibleState } from "vscode";
+import { ext } from "../../extensionVariables";
+import { localize } from "../../utils/localize";
+import { treeUtils } from "../../utils/treeUtils";
+import type { ContainerAppModel } from "../ContainerAppItem";
+import { ScaleItem } from "../scaling/ScaleItem";
+import type { RevisionsItemModel } from "./RevisionItem";
+
+export class RevisionDraftItem implements RevisionsItemModel {
+    static readonly idSuffix: string = 'revisionDraft';
+    static readonly contextValue: string = 'revisionDraftItem';
+    static readonly contextValueRegExp: RegExp = new RegExp(RevisionDraftItem.contextValue);
+
+    id: string;
+    revisionsMode: KnownActiveRevisionsMode;
+
+    constructor(readonly subscription: AzureSubscription, readonly containerApp: ContainerAppModel, readonly revision: Revision) {
+        this.id = `${this.containerApp.id}/${RevisionDraftItem.idSuffix}`;
+        this.revisionsMode = containerApp.revisionsMode;
+    }
+
+    get revisionName(): string {
+        return nonNullProp(this.revision, 'name');
+    }
+
+    /**
+     * @example gets "rev-1" of "my-app--rev-1"
+     */
+    private get baseRevisionName(): string {
+        return this.revisionName.substring(this.revisionName.lastIndexOf('--') + 2);
+    }
+
+    static hasDescendant(item: RevisionsItemModel): boolean {
```

Comment:
What's this for? I see the code that uses it is currently commented out, but I don't totally understand what that's doing either.

Reply:
`RevisionDraftItem.hasDescendant()` checks to see if the passed item is a child of the `RevisionDraftItem`.  Since the item has to have a `RevisionsItemModel` and must match the revision base draft, we can conclude it's a child of the draft item (actually I just realized there's an edge case where this doesn't make sense - if you pass `RevisionDraftItem` to itself... but that's just kind of weird to do anyway...).

One example where this method is useful - technically every revision in multiple revisions mode has a `ScaleItem` child.  We don't want all of them to show unsaved changes when we detect a diff, we only care about showing those changes when the child is a descendant of the draft item.  This method will be useful for future children as well.  If you prefer we could always make this a utility function rather than a static method.  I just kind of liked reading: `RevisionDraftItem.hasDescendant(this)` because it felt like a cool way to describe exactly what is being accomplished.

Reply:
Actually, I just realized why this might be confusing.  Even though reading off the active revision draft file works, I think it makes more intuitive sense to do:

`return item.revision.name === this.revision.name;` or similar.

Reply:
It would probably be more intuitive to read if it said `return item.revision.name === this.revision.name;`.  Unfortunately there's not context of `this` in the static method, so we have to fetch the revision draft file to get this info.

Reply:
Would it make more sense for the `ScaleItem` to have a `isDescendant` function then? And that could probably be some sort of shared function on an abstract base class or an interface or something. Then this could just call `ScaleItem.isDescendant(this)`.

I'm not entirely sure, but I can imagine that the behavior of what is a descendant might change as we add more nodes (like the containers node) depending on how deep it can drill down.

Also not entirely sure when this is being used, so if that wouldn't work, disregard.

Reply:
That could work, but my fear is that enforcing inheritance here will make it more painful to maintain down the road.  I personally think the best options are a) either a static method on `RevisionDraftItem` or b) a separate util method if it seems unintuitive to couple the method to the draft item

Reply:
> I'm not entirely sure, but I can imagine that the behavior of what is a descendant might change as we add more nodes (like the containers node) depending on how deep it can drill down.

How do you feel about me refactoring to this method if we find that we're having to special case the node behaviors later down the line.  I think it's really unlikely that I'll end up having to do this, but I can't say with 100% certainty.  I I don't think refactoring to this method wouldn't be very expensive to do later if the need arises

Reply:
Yeah sure. We can address the problem (if it exists) when we actually have more children/complicated scenarios.